src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -0,0 +1,544 @@
+import org.openstreetmap.atlas.tags.SyntheticNearestNeighborCountryCodeTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Iterables;
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.TopologyException;
+
+/**
+ * The {@link RawAtlasCountrySlicer} consumes a raw Atlas and produces a fully country-sliced Atlas
+ * as output.
+ *
+ * @author mgostintsev
+ */
+public class RawAtlasCountrySlicer
+{
+    private static final Logger logger = LoggerFactory.getLogger(RawAtlasCountrySlicer.class);
+
+    // JTS converters
+    private static final JtsPolygonConverter JTS_POLYGON_CONVERTER = new JtsPolygonConverter();
+    private static final JtsPolyLineConverter JTS_POLYLINE_CONVERTER = new JtsPolyLineConverter();
+    private static final JtsLocationConverter JTS_LOCATION_CONVERTER = new JtsLocationConverter();
+
+    // The raw Atlas to slice
+    private Atlas rawAtlas;
+
+    // The countries we're interested in slicing against
+    private final Set<IsoCountry> countries;
+
+    // Contains boundary MultiPolygons
+    private final CountryBoundaryMap countryBoundaryMap;
+
+    private RawAtlasChangeSet afterSlicingLinesAndPoints = new RawAtlasChangeSet();
+    private final RawAtlasChangeSet afterSlicingRelations = new RawAtlasChangeSet();
+
+    // Mapping between Coordinate and created Temporary Point identifiers. This is to avoid
+    // duplicate points at the same locations and to allow fast lookup to construct new lines
+    // requiring the temporary point as a Line shape point
+    private final Map<Coordinate, Long> newPointCoordinates = new HashMap<>();
+
+    private final RawAtlasSlicingStatistic statistics = new RawAtlasSlicingStatistic(logger);
+
+    public RawAtlasCountrySlicer(final Atlas rawAtlas, final Set<IsoCountry> countries,
+            final CountryBoundaryMap countryBoundaryMap)
+    {
+        this.rawAtlas = rawAtlas;
+        this.countries = countries;
+        this.countryBoundaryMap = countryBoundaryMap;
+    }
+
+    /**
+     * TODO Complete once Relation slicing is implemented
+     *
+     * @return a country-sliced {@link Atlas}
+     */
+    public Atlas slice()
+    {
+        logger.info("Starting country slicing Atlas {}", this.rawAtlas.getName());
+
+        // Slice lines and points
+        sliceLines();
+        slicePoints();
+
+        // Apply changes and rebuild the Atlas with the changes before slicing Relations
+        final RawAtlasChangeSetBuilder changeBuilder = new RawAtlasChangeSetBuilder(this.rawAtlas,
+                this.afterSlicingLinesAndPoints);
+        final Atlas withSlicedWaysAndPoints = changeBuilder.applyChanges();
+        this.rawAtlas = withSlicedWaysAndPoints;
+
+        // Clear for GC
+        this.afterSlicingLinesAndPoints = null;
```

Comment:
Still not sure if this is necessary... When the `RawAtlasCountrySlicer` is dereferenced, this object will be cleaned automatically too.